### ISSUE TITLE
Exception after a dialogue reaches certain length

### ISSUE THREAD

**Issue by al42and** [petals-infra/chat.petals.dev#3]
After a long convo, I'm getting the following error:

```
Request failed. Retry
Traceback (most recent call last):
  File "/home/borzunov/chat.petals.ml/app.py", line 127, in generate
    outputs = model.generate(
  File "/home/borzunov/.local/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/borzunov/.local/lib/python3.10/site-packages/petals/client/remote_generation.py", line 182, in generate
    hidden_state = session.step(hidden_state, prompts=intermediate_prompts, hypo_ids=hypo_ids)[:, -1]
  File "/home/borzunov/.local/lib/python3.10/site-packages/petals/client/inference_session.py", line 233, in step
    raise ValueError(
ValueError: Maximum length exceeded: prefix 0 + current 784 exceeds pre-allocated maximum 768
```

A more helpful error message, preferably with a button to reset the conversation, would be better.

<details><summary>Screenshot of the error</summary>

<p>

![image](https://user-images.githubusercontent.com/933873/212572257-f3803013-2b6c-4038-9be9-f2e008cd5ec2.png)

</p>
</details>

**Comment by justheuristic** [comment]
Related: https://github.com/bigscience-workshop/petals/issues/146

**Comment by DarkIlluminatus** [comment]
it seems like the dialogue length is exceeding the maximum limit set in the source code. You might want to consider modifying this value to accommodate longer dialogues. However, do keep in mind that very long dialogues might have other implications on performance and resource usage. Good luck!

**Comment by borzunov** [comment]
Hi @al42and @DarkIlluminatus,

Sorry for the long response - we've addressed this issue a few months ago but didn't respond here.

We increased the inference session length limit to 8192 tokens for the latest models like Llama 2 (70B), Stable Beluga 2, Falcon 180B. This should be enough for most cases - Llama 2 is trained on sequences with up to 4096 tokens, so it's unlikely to perform well on sequences much longer than 8192 tokens anyway.

Also, we updated this chat app to [gradually increase](https://github.com/petals-infra/chat.petals.dev/blob/main/static/chat.js#L186) allocated memory to save resources: we allocate an inference session for 512 tokens, then expand it to 2048 tokens when needed, then expand to 8192.

Let us know if you have other issues!
